src/executionFiles/bridges/nxtp.execute.ts

```diff
@@ -240,4 +270,46 @@ export class NXTPExecutionManager {
     nxtpSDK.removeAllListeners()
     return status
   }
+
+  private calculateRelayerFee = async (
+    nxtpSDK: NxtpSdk,
+    preparedTransaction: ReceiverTransactionPreparedPayload
+  ): Promise<string> => {
+    let calculateRelayerFee = '0'
+
+    const chainIdsForPriceOracle = getDeployedChainIdsForGasFee()
+
+    if (
+      chainIdsForPriceOracle.includes(
+        preparedTransaction.txData.receivingChainId
+      )
+    ) {
+      const gasNeeded = await nxtpSDK.estimateMetaTxFeeInReceivingToken(
+        preparedTransaction.txData.sendingChainId,
+        preparedTransaction.txData.sendingAssetId,
+        preparedTransaction.txData.receivingChainId,
+        preparedTransaction.txData.receivingAssetId
+      )
+
+      calculateRelayerFee = gasNeeded.toString()
+    }
+
+    return calculateRelayerFee
+  }
+
+  private decryptData = async (
+    decryptHook: DecryptHook,
+    { txData, encryptedCallData }: ReceiverTransactionPreparedPayload
+  ): Promise<string> => {
+    let callData = '0x'
+    if (txData.callDataHash !== utils.keccak256(callData)) {
+      try {
+        callData = await decryptHook(encryptedCallData)
+      } catch (e) {
+        // TODO: update process failed
```

Comment:
We may not want to catch the error here and instead handle it outside

Reply:
Ah yeah forgot about that todo.
Don't we want to set the process to failed here?
That's also how we do it above